eZ/Publish/Core/Persistence/Legacy/User/Handler.php

```diff
@@ -166,38 +168,82 @@ public function delete($userId)
     }
 
     /**
-     * Create new role.
+     * Create new role draft.
      *
-     * @param \eZ\Publish\SPI\Persistence\User\Role $role
+     * Sets status to Role::STATUS_DRAFT on the new returned draft.
+     *
+     * @param \eZ\Publish\SPI\Persistence\User\RoleCreateStruct $createStruct
+     *
+     * @return \eZ\Publish\SPI\Persistence\User\Role
+     */
+    public function createRole(RoleCreateStruct $createStruct)
+    {
+        return $this->internalCreateRole($createStruct);
+    }
+
+    /**
+     * Creates a draft of existing defined role.
+     *
+     * Sets status to Role::STATUS_DRAFT on the new returned draft.
+     *
+     * @param mixed $roleId
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException If role with defined status is not found
+     *
+     * @return \eZ\Publish\SPI\Persistence\User\Role
+     */
+    public function createRoleDraft($roleId)
+    {
+        $createStruct = $this->mapper->createCreateStructFromRole(
+            $this->loadRole($roleId)
+        );
+
+        return $this->internalCreateRole($createStruct, $roleId);
+    }
+
+    /**
+     * Internal method for creating Role.
+     *
+     * Used by self::createRole() and self::createRoleDraft()
+     *
+     * @param \eZ\Publish\SPI\Persistence\User\RoleCreateStruct $createStruct
+     * @param mixed|null $roleId Used by self::createRoleDraft() to retain Role id in the draft
      *
      * @return \eZ\Publish\SPI\Persistence\User\Role
      */
-    public function createRole(Role $role)
+    protected function internalCreateRole(RoleCreateStruct $createStruct, $roleId = null)
     {
+        $createStruct = clone $createStruct;
+        $role = $this->mapper->createRoleFromCreateStruct(
+            $createStruct
+        );
+        $role->id = $roleId;
```

Comment:
Ok, I see now that is done in the gateway, ignore the ID part :)
But it should be tested anyway.